src/set_theory/game/nim.lean

```diff
@@ -1,33 +1,37 @@
 /-
 Copyright (c) 2020 Fox Thomson. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
-Authors: Fox Thomson
+Authors: Fox Thomson, Markus Himmel
 -/
+import data.nat.bitwise
 import set_theory.game.impartial
-import set_theory.ordinal
-import data.set
-import logic.basic
-
-universes u v
+import set_theory.ordinal_arithmetic
 
 /-!
 # Nim and the Sprague-Grundy theorem
 
 This file contains the definition for nim for any ordinal `O`. In the game of `nim O₁` both players
 may move to `nim O₂` for any `O₂ < O₁`.
 We also define a Grundy value for an impartial game `G` and prove the Sprague-Grundy theorem, that
-`G` is equivalent to `nim (Grundy_value G)`.
+`G` is equivalent to `nim (grundy_value G)`.
+Finally, we compute the sum of finite Grundy numbers: if `G` and `H` have Grundy values `n` and `m`,
+where `n` and `m` are natural numbers, then `G + H` has the Grundy value `n xor m`.
 
 ## Implementation details
 
-The pen-and-paper definition of nim defines the possible moves of `nim O` to be `{ O' | O' < O}`.
+The pen-and-paper definition of nim defines the possible moves of `nim O` to be `{ O' | O' < O }`.
```

Comment:
```suggestion
The pen-and-paper definition of nim defines the possible moves of `nim O` to be `{O' | O' < O}`.
```
I think we only use extra spaces around braces when they surround a list of tactics. 

Alternatively, this is pseudo-Lean and perhaps it should actually be LaTeX?
```suggestion
The pen-and-paper definition of nim defines the possible moves of `nim O` to be $\{ O' | O' < O\}$.
```